### PR TITLE
feat: Rename `wasmer-engine-jit` to `wasmer-engine-universal`

### PR DESCRIPTION
**Pull request by Hywan**
This (small) patch renames the `wasmer-engine-jit` crate into
`wasmer-engine-universal`. We believe the new name conveys
better the intent of the engine.

I will update the `CHANGELOG` later.

### CODE REVIEW

**Comment by Hywan** [comment]
bors try

File: lib/engine-universal/src/artifact.rs [hunk 1]
```diff
@@ -253,12 +257,12 @@ impl JITArtifact {
 
     /// Get the default extension when serializing this artifact
     pub fn get_default_extension(_triple: &Triple) -> &'static str {
-        // `.wjit` is the default extension for all the triples
-        "wjit"
+        // `.wuniversal` is the default extension for all the triples
+        "wuniversal"
```
Comment (Hywan) [inline lib/engine-universal/src/artifact.rs:261]: This part is important.

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors cancel

**Comment by Hywan** [comment]
bors try

**Review by syrusakbary — APPROVED** [review]

**Comment by Hywan** [comment]
bors try

File: examples/engine_headless.rs [hunk 2]
```diff
@@ -8,7 +8,7 @@
 //! What problem does it solve, and what does it mean?
 //!
 //! Once a Wasm module is compiled into executable code and stored
-//! somewhere (e.g. in memory with the JIT engine, or in a native
+//! somewhere (e.g. in memory with the Universal engine, or in a native
 //! object with the native engine), the module can be instantiated and
```
Comment (jubianchi) [inline examples/engine_headless.rs:12]:
```suggestion
//! shared object file with the native engine), the module can be instantiated and
```

File: lib/api/README.md [hunk 3]
```diff
@@ -1,7 +1,8 @@
 # `wasmer` [![Build Status](https://github.com/wasmerio/wasmer/workflows/build/badge.svg?style=flat-square)](https://github.com/wasmerio/wasmer/actions?query=workflow%3Abuild) [![Join Wasmer Slack](https://img.shields.io/static/v1?label=Slack&message=join%20chat&color=brighgreen&style=flat-square)](https://slack.wasmer.io) [![MIT License](https://img.shields.io/github/license/wasmerio/wasmer.svg?style=flat-square)](https://github.com/wasmerio/wasmer/blob/master/LICENSE) [![crates.io](https://img.shields.io/crates/v/wasmer.svg)](https://crates.io/crates/wasmer)
 
-[`Wasmer`](https://wasmer.io/) is the most popular [WebAssembly](https://webassembly.org/)
-runtime for Rust (...and also the fastest). It supports JIT (Just in Time) and AOT (Ahead of time)
+[`Wasmer`](https://wasmer.io/) is the most popular
+[WebAssembly](https://webassembly.org/) runtime for Rust (...and also
+the fastest). It supports JIT (Just in Time) and AOT (Ahead of time)
```
Comment (jubianchi) [inline lib/api/README.md:5]: Now that engines have more generic names, should we change this?

```suggestion
the fastest). It supports JIT (Just in Time, via the Universal engine) and AOT (Ahead of time, via the Native Shared Object engine)
```

File: lib/c-api/src/wasm_c_api/engine.rs [hunk 4]
```diff
@@ -65,9 +65,9 @@ impl Default for wasmer_compiler_t {
 #[repr(C)]
 #[allow(non_camel_case_types)]
 pub enum wasmer_engine_t {
-    /// Variant to represent the JIT engine. See the
-    /// [`wasmer_engine_jit`] Rust crate.
-    JIT = 0,
+    /// Variant to represent the universal engine. See the
```
Comment (jubianchi) [inline lib/c-api/src/wasm_c_api/engine.rs:68]:
```suggestion
    /// Variant to represent the Universal engine. See the
```

File: lib/cli/README.md [hunk 5]
```diff
@@ -35,7 +35,7 @@ The Wasmer supports the following features:
 * `cranelift`: support for the [Cranelift compiler].
 * `llvm`: support for the [LLVM compiler].
 
-[JIT Engine]: https://github.com/wasmerio/wasmer/tree/master/lib/engine-jit/
+[Universal Engine]: https://github.com/wasmerio/wasmer/tree/master/lib/engine-universal/
```
Comment (jubianchi) [inline lib/cli/README.md:38]:
```suggestion
[Universal engine]: https://github.com/wasmerio/wasmer/tree/master/lib/engine-universal/
```

File: lib/engine-universal/src/artifact.rs (hunk 1 above)
Comment (syrusakbary) [reply lib/engine-universal/src/artifact.rs:261]: I don't like the extension. I think a better name is `wasmu` or `wuni` (I incline more for `wasmu`)

**Comment by Hywan** [comment]
bors try

File: lib/api/README.md (hunk 3 above)
Comment (Hywan) [reply lib/api/README.md:5]: I think we should make a final pass on the `README.md` after all the renamings.

File: lib/engine-universal/src/artifact.rs (hunk 1 above)
Comment (Hywan) [reply lib/engine-universal/src/artifact.rs:261]: Go for `.wasmu` :-).

**Comment by Hywan** [comment]
It didn't failed, I've cancelled it myself.

bors r+

**Pull request merged**